# 들어가기 : 주석 

# 표준 입력과 표준 출력

In [4]:
12345
10
#ipython 라이브러리의 기본 기능
# =print(12345) 표준 출력 기능을 내장하고 있다.

10

In [5]:
print(12345)
10

12345


10

In [6]:
'12345'

'12345'

In [7]:
type('12345')

str

In [8]:
type(12345)

int

In [9]:
123456+10

123466

In [12]:
#'12345'+10 #type error

In [13]:
type("12345")

str

In [14]:
"12345"

'12345'

In [15]:
"12345"*3

'123451234512345'

In [19]:
age = input('당신의 나이을 입력하세요')
print (age)

당신의 나이을 입력하세요 25


25


In [20]:
age

'25'

In [22]:
name = input('당신의 이름을 입력하세요.') #입력 받은 이름
print (name)                             #이름 출력

당신의 이름을 입력하세요. 상준


상준


In [23]:
name

'상준'

In [25]:
print(age, name)

25 상준


In [31]:
print( '나는 '+name+'이고 나이는 '+age+'입니다')

나는 상준이고 나이는 25입니다


In [36]:
int(age)+10

35

In [38]:
age = int(age)

In [39]:
age + 10

35

In [41]:
age = int(input('input your age'))
age

input your age 25


25

# 연습

In [45]:
# 자기 이름을 삼행시 짓기

print('박: 박수칠때 떠나는')
print('상: 상상을 하는')
print('준: 준비된 사람')

박: 박수칠때 떠나는
상: 상상을 하는
준: 준비된 사람


In [53]:
# 자기이름으로 삼행시 짓기 
a = input('박')
b = input('상')
c = input('준')
print(a, b, c)

박 박수칠 때 떠나는 멋진 남자가 되는
상 상상을 하는 
준 준비된 남자


박수칠 때 떠나는 멋진 남자가 되는 상상을 하는  준비된 남자


In [54]:
print(a, b, c, sep='\n')

박수칠 때 떠나는 멋진 남자가 되는
상상을 하는 
준비된 남자


문자열의 시작이 \ 역슬래시 -> 이스케이프 문자, 특수 문자
\n
\t
\\

In [64]:
삼행시 = a +'\n' + b +'\n' + c +'\n'
print(삼행시)

박수칠 때 떠나는 멋진 남자가 되는
상상을 하는 
준비된 남자



# 변수

변수 type safe
변수를 선언할 때 타입(유형)을 지정하지 않아도 되는 특성
변수명 = 값

[변수의 명명 규칙]
1. 문자, 숫자, 밑줄을 가지고 만든다
2. 숫자로 시작하면 x
3. 대소문자를 구분
4. 예약어, 특수기호, 공백 x

In [69]:
false = 10

In [70]:
x = 10
y = 'hello'
z = 3.141592
x,y,z

(10, 'hello', 3.141592)

In [72]:
print(type(x), type(y), type(z))

<class 'int'> <class 'str'> <class 'float'>


In [75]:
a,b,c =10,20,30 #unfacking #a=10; b=20; c=30
a,b,c 

(10, 20, 30)

In [76]:
a = b = c = 40
a,b,c

(40, 40, 40)

In [77]:
d = True #bool형
e = None #type이 없음 Nonetype
d,e

(True, None)

In [1]:
a = 10; b = a
print(a,b)
id(a), id(b)

10 10


(140714362940120, 140714362940120)

In [83]:
a = 20 
print(a,b)

20 10


In [84]:
id(a), id(b)

(140714362940440, 140714362940120)

In [2]:
aa = 10; bb = 10;
print(aa,bb)
id(aa), id(bb)

10 10


(140714362940120, 140714362940120)

In [3]:
#의미 부여 하기 어려운 변수명 > 수식이 아닌 경우는 사용하지 않음
x = 100
y = 100

# 의미 있는 변수명 
age = 20
name = '김영리'
user_name = 'Alic'
admin_name = 'Bob'

#대문자 Upper-case 상수에 사용
PI = 3.141592
MAX_CONNECTIONS = 10 #최대 연결 수
MAX_VALUE = 0 #최솟값


[NAME_SPACE] 

변수가 최초로 정의된 영역에 따라 유효범위가 결정됩니다.

- 지역변수 : 함수 영역 내에서 선언한 변수
- 전역변수 : 함수 영역 밖에서 선언한 변수, global 키워드
- 정적변수 : 클래스에서 사용하는 변수

In [12]:
#지역변수
def local_example():
    local_var = 100
    print('여기는 함수의 영역입니다. '+ local_var)

#local_var #NameError: name 'Local_var' is not defined

In [26]:
#전역 변수
var_test = '나는 전역이다'
print(var_test)

def global_example():
    var_test = 100
    print('여기는 함수의 영역입니다. ', var_test)

global_example()
print(var_test)

나는 전역이다
여기는 함수의 영역입니다.  100
나는 전역이다


# 연습2

In [31]:
#변수 영역 연습

counter = 0 #global

def increment_counter():
    global counter
    counter += 1
    print(f"카운터: {counter}")

increment_counter() #출력: 카운터 1
increment_counter() #출력: 카운터 2
increment_counter() #출력: 카운터 3

카운터: 1
카운터: 2
카운터: 3


In [41]:
a= (2,3,4) ; a=b
a is b

True

In [45]:
a=[1,2,3]
id(a)

1968665166080

In [46]:
a=[1,2,3]
b = a

In [47]:
a is b

True

In [48]:
a[1]=4
a

[1, 4, 3]

In [49]:
b

[1, 4, 3]

In [52]:
a = [1,2,3]
b = a[:]
a[1] = 4
a
print(a,b)

[1, 4, 3] [1, 2, 3]


In [53]:
from copy import copy
a = [1,2,3]
b = copy(a)

In [56]:
b is a

False

In [57]:
a, b=('python','life')

In [58]:
a =3 
b = 5
a,b = b,a
print(a,b)

5 3


# 초보자를 위한 파이썬 300제

In [60]:
print("Hello World")

Hello World


In [61]:
print("Mary's cosmetics")

Mary's cosmetics


In [62]:
print('신씨가 소리질렀다. "도둑이야".')

신씨가 소리질렀다. "도둑이야".


In [66]:
#print("C:\windows")

In [67]:
print("안녕하세요.\n만나서\t\t반갑습니다.") #\n은 줄바꿈, \t는 들여쓰기

안녕하세요.
만나서		반갑습니다.


In [68]:
#오늘은 일요일
print("오늘은", "일요일")

오늘은 일요일


In [72]:
print("naver", "kakao","sk", "samsung", sep="/")

naver/kakao/sk/samsung


In [74]:
print("first", end=""); print("second")

firstsecond


In [76]:
print(5/3)

1.6666666666666667


In [78]:
#삼성전자라는 변수로 50,000원을 바인딩해보세요. 삼성전자 주식 10주를 보유하고 있을 때 총 평가금액을 출력하세요.
삼성전자 = 50000
총평가금액 = 삼성전자 * 10
print(총평가금액)

500000


In [80]:
시가총액 = 298000000000000
현재가 = 50000
PER = 15.79
print(시가총액, type(시가총액))
print(현재가, type(현재가))
print(PER, type(PER))

298000000000000 <class 'int'>
50000 <class 'int'>
15.79 <class 'float'>


In [81]:
s = 'hello'
t = 'python'
print(s + '!' ,t)

hello! python


In [82]:
print(2+2*3)

8


In [84]:
a="132"
print(type(a))

<class 'str'>


In [87]:
num_str="720"
num_int = int(num_str)
print(num_int+1, type(num_int))

721 <class 'int'>


In [89]:
data = "15.79"
data = float(data)
print(data, type(data))

15.79 <class 'float'>


In [91]:
year = "2020"
print(int(year)-3)
print(int(year)-2)
print(int(year)-1)

2017
2018
2019


In [93]:
월 = 48584
총금액 = 월 * 36
print(총금액)

1749024


# 자료형

기본 자료형 : 숫자형, 문자열형 str, bool형
복합 자료형 : 리스트, 셋, 튜플, 딕셔너리

|type|설명|예|
|--------|--------|--------|
|정수형(int)| 정수를 저장하는 자료형 | 1, -5, 100|
|실수형(float)| 소수점이 있는 숫자를 저장하는 자료형 | 3.14, -0.001, 2.0|
|문자열(str)| 텍스트를 저장하는 자료형 | "Hello", "파이썬"|
|논리형(bool)| 참/거짓 값을 저장하는 자료형 | True, False|

- 수정 가능한 데이터 타입 : 리스트, 딕녀너리, 셋
- 수정 불가능한 데이터 타입 : 수치형, 부울형, 문자열, 튜플

In [95]:
#수치형 -int
20, type(20)

(20, int)

In [96]:
age = 20
type(age)

int

1. 진법 : 10진수, 2진수, 8진수, 16진수
2. 입력하는 방법은 다르지만, print() 결과는 동일하다.

In [97]:
a = 10
b = 0b0101 #2진수 - 0/1 로 표현하는 방법 -> 0101 = 2^2 + 2^0
a,b

(10, 5)

In [100]:
0b00010001110010

1138

In [105]:
c=0o10 #0o 8진수 10 -> 8^1 + 8^0 =8*1 + 0*0 = 8
c

8

In [103]:
d=0b1000
d

8

In [109]:
e = 0x16d #1*16^2 + 6* 16^1 + 13*16*0 =256+96+1
e

365

In [ ]:
지수의 표기
숫자e2 -> 10^2 -> 10**2
숫자e-2 -> 10^-2 -> 10**-2

In [112]:
fa = 10e2
fa, 10**2

1000.0

In [114]:
fb =10e-2
fb

0.1

In [115]:
fc = 3.1415e9
fc

3141500000.0

BOOLEAN값 - TRUE, FALSE
논리연산, 비교연산

ba = True
bb = true        #값이 없는 변수이기 떄문
type(ba), type(bb)

In [120]:
type(ba), ba, ba == 1, ba==0

(bool, True, True, False)

In [121]:
ba == 'True'

False

#논리 연산자 - and or not

bc =1 
ba and bc

In [127]:
# 사칙연산 + - * / // % 
a = 2
b = 3
a+b, a-b, a*b, a/b, a%b ,a//b #나머지, 몫

(5, -1, 6, 0.6666666666666666, 2, 0)

In [132]:
#비교 연산자
1+2 == 3, 1+2 != 3 , 1 + 3 > 3 , 1+3<=4

(True, False, True, True)

In [135]:
#비교 연산자와 논리 연산자의 결합
10%2 == 0 and 12%3 == 0

True

In [136]:
a = '1234'
b = '5678'
a,b, id(a), id(b)

('1234', '5678', 1968627963616, 1968667225440)

In [138]:
#대입연산자
a = 10
b = a + 2
a,b

(10, 12)

In [218]:
a= 20
a+=1  #a = a + 1
a,a,a,a,a,a,a

(21, 21, 21, 21, 21, 21, 21)

In [219]:
a -=1     #a = a - 1, 21 -1 =20
a

20

In [232]:
#quiz 
a = 3
b = 2
a += 2 + b
a

7

In [ ]:
#문자열 str 
' ', " ", """  """ , ''' ''' docstring

In [233]:
a = 'hello'
b = 'python'
a+b

'hellopython'

In [234]:
c = '*'
c*10


'**********'

In [236]:
e = '1'
e == '1'

True

In [237]:
#문자열은 리스트, 열거형
a = '20010331Rabcd'
a[0] #숫자 자리수 > 인덱스

'2'

In [239]:
a[3] #인덱싱

'1'

In [244]:
a[:4] #영역을 지정할 떄 : (콜론), 슬라이싱 slice

'2001'

In [246]:
#index = 0에서 시작
a[8:]

'Rabcd'

In [247]:
s = 'abc'
a = s
a is s

True

In [248]:
b =s[:]
a is b

True

In [250]:
s1 = '1'*1000
s1

'111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111

In [251]:
b =s1[:]
s1 == b

True

1. 유니코드 문자표준
2. 영어는 ascii -> 유니코드
3. 한국어도 유니코드
4. 유니코드가 글자의 크기에 따라 utf-8, [utf-16,utf-32]
5. 파이썬의 문자열 str -> unicode 기반이다.
6. euc-kr 예전에 우리나라에서 쓰던 한글코드
7. cp949 :예전에 MS 윈도우즈의 한글코드

In [5]:
text = "파이썬"
print(text)
print( ord('파') )

파이썬
54028


In [18]:
#인코딩
enc_text = text.encode('utf-8')
enc_text

b'\xed\x8c\x8c\xec\x9d\xb4\xec\x8d\xac'

In [13]:
#디코딩
dec_text = enc_text.decode('utf-8')
dec_text

'파이썬'

1. 파이썬의 인덱스는 0부터 시작
2. 음수 인덱스 : 뒤에서부터 -1

In [20]:
''' docstring \n
start \n
end \n
'''

' docstring \n\nstart \n\nend \n\n'

In [22]:
doc_string = '''docstring \n
start \n
end \n
'''
print(doc_string)
doc_string

docstring 

start 

end 




'docstring \n\nstart \n\nend \n\n'

In [23]:
type(text)

str

In [24]:
#str의 함수 

dir()

['In',
 'Out',
 '_',
 '_13',
 '_14',
 '_15',
 '_16',
 '_17',
 '_18',
 '_19',
 '_20',
 '_22',
 '_23',
 '_6',
 '_7',
 '_8',
 '__',
 '___',
 '__builtin__',
 '__builtins__',
 '__doc__',
 '__loader__',
 '__name__',
 '__package__',
 '__session__',
 '__spec__',
 '_dh',
 '_i',
 '_i1',
 '_i10',
 '_i11',
 '_i12',
 '_i13',
 '_i14',
 '_i15',
 '_i16',
 '_i17',
 '_i18',
 '_i19',
 '_i2',
 '_i20',
 '_i21',
 '_i22',
 '_i23',
 '_i24',
 '_i3',
 '_i4',
 '_i5',
 '_i6',
 '_i7',
 '_i8',
 '_i9',
 '_ih',
 '_ii',
 '_iii',
 '_oh',
 'dec_text',
 'doc_string',
 'enc_text',
 'exit',
 'get_ipython',
 'open',
 'quit',
 'text']

In [25]:
dir(text)

['__add__',
 '__class__',
 '__contains__',
 '__delattr__',
 '__dir__',
 '__doc__',
 '__eq__',
 '__format__',
 '__ge__',
 '__getattribute__',
 '__getitem__',
 '__getnewargs__',
 '__getstate__',
 '__gt__',
 '__hash__',
 '__init__',
 '__init_subclass__',
 '__iter__',
 '__le__',
 '__len__',
 '__lt__',
 '__mod__',
 '__mul__',
 '__ne__',
 '__new__',
 '__reduce__',
 '__reduce_ex__',
 '__repr__',
 '__rmod__',
 '__rmul__',
 '__setattr__',
 '__sizeof__',
 '__str__',
 '__subclasshook__',
 'capitalize',
 'casefold',
 'center',
 'count',
 'encode',
 'endswith',
 'expandtabs',
 'find',
 'format',
 'format_map',
 'index',
 'isalnum',
 'isalpha',
 'isascii',
 'isdecimal',
 'isdigit',
 'isidentifier',
 'islower',
 'isnumeric',
 'isprintable',
 'isspace',
 'istitle',
 'isupper',
 'join',
 'ljust',
 'lower',
 'lstrip',
 'maketrans',
 'partition',
 'removeprefix',
 'removesuffix',
 'replace',
 'rfind',
 'rindex',
 'rjust',
 'rpartition',
 'rsplit',
 'rstrip',
 'split',
 'splitlines',
 'startswith',
 'stri

In [35]:
text.__add__("4")

'파이썬4'

In [31]:
text.count('t')

0

In [37]:
text + "4"

'파이썬4'

In [42]:
text.__hash__()

8026764794004708901

excape 문자 
\ '역슬래시'

In [47]:
print('hello \n py\'') #excape 문자는 특수문자를 그냥 텍스트로 인식하게 한다.

hello 
 py'


In [50]:
text.__mul__(4)

'파이썬파이썬파이썬파이썬'

In [ ]:
raw-string : 모두 다 문자열이다 r'str?\'
f-string : 형기 코드가 내폰된 문자열이다 f'str{text}'

In [52]:
str_1 = '12\\4' #'12\x04' \는 escape 문자 > 무시하는 방법은 r
str_1

'12\\4'

In [53]:
raw_str = r'12\4'
raw_str

'12\\4'

In [56]:
raw_str_1 = r'Hello\nWorld'
print(raw_str_1)
raw_str_1

Hello\nWorld


'Hello\\nWorld'

In [64]:
f' {수식} 문자열이에요 {변수}'

NameError: name '수식' is not defined

In [67]:
name = 'sj'
age =25
f_str_1 = f'제이름은 {name} 내 나이는 {age} 입니다.'
f_str_1

'제이름은 sj 내 나이는 25 입니다.'

In [71]:
# 이름과 자기소개를 입력받아서 출력하는 프로그램 작성
name = input('이름')
age = input('나이')
f_str_1 = f'제 이름은 {name} 내 나이는 {age} 입니다.'
print(f_str_1)

이름 상준
나이 25


제 이름은 상준 내 나이는 25 입니다.


In [75]:
#dir(f_str_1)
format_str = ' {0:.2f} {1:s} are worth us$ {2:d} 입니다'  #template
format_str.format(99.99, 'dollars',100)

' 99.99 dollars are worth us$ 100 입니다'

In [92]:
menu = input('주문할 메뉴를 입력하세요')
price = input('한 개의 가격이 얼마인가요?')
count = float( input('몇 개나 주문하세요?'))

total = float(price)*float(count)
format_str = ' {0:} 메뉴의 {1:2f}개 가격은 총 {2:.0f}입니다.'
format_str.format(menu, count, total)

주문할 메뉴를 입력하세요 불
한 개의 가격이 얼마인가요? 2000
몇 개나 주문하세요? 2


' 불 메뉴의 2.000000개 가격은 총 4000입니다.'

In [1]:
items = [
    {"name": "아메리카노", "price": 3000, "qty": 2},
    {"name": "카페라떼", "price": 4000, "qty": 1},
    {"name": "샌드위치", "price": 5500, "qty": 3}
]
header_str = '{0:10}{1:10}{2:10}{3:10}' #인덱스 : 데이터 길이
print(header_str.format('상품명', '가격', '수량', '총액'))
print('-'*50)

data_str = '{0:10}{1:10}원{2:10}개{3:10}원'
for item in items : 
    item_name = item['name']
    item_price = item['price']
    item_qty = item['qty']    
    total_price = item_price * item_qty
    print(data_str.format(item_name, item_price, item_qty, total_price))

상품명       가격        수량        총액        
--------------------------------------------------
아메리카노           3000원         2개      6000원
카페라떼            4000원         1개      4000원
샌드위치            5500원         3개     16500원


## TypeError, ValueError

In [59]:
b = 2
c = '3'
#b+c #typeerror
b + int(c)

5

In [62]:
c = '3a'
b + int(c[:1]) #ValueError

5